### DISCUSSION TITLE
Breaking numbers out of an array

### DISCUSSION THREAD

**Discussion by memark** [uiua-lang/uiua#604] [Learning] [answered]
Uiua 0.13.0.

I often find the need to break numbers out of a (small) array, e.g. converting `[a b c]` to `a b c` (or `c b a`, doesn't matter).

I first thought that `un couple` would help me:

```
$ uiua eval "°⊟ ⊟ 1 2"
2
1
```

but that doesn't work with 3 arguments.

Next I turned to `un join`, but the inverse surprised me:

```
$ uiua eval "°(⊂⊂) ⊂⊂ 1 2 3"
[3]
[2]
1
```

The best I've come up with is

```
$ uiua eval "⊃(⊡0|⊡1|⊡2) ⊂⊂ 1 2 3"
3
2
1
```

but it feels cumbersome.

Is there a more idiomatic way?

**Comment by kaikalii** [comment]
See the [Inverses tutorial](https://www.uiua.org/tutorial/inverses#un)

**Reply by memark** [reply, marked as answer]
Thanks! `°[⊙⊙∘]` solved my problem.

**Reply by Omnikar** [reply]
[The idiomatic way to do this is now `°⊟₃` instead.](https://uiua.org/pad?src=0_14_0-dev_3__wrDiip_igoMgWzEgMiAzXQo=)

**Reply by memark** [reply]
That's even clearer! Thanks, @Omnikar.

**Reply by haydenflinner** [reply]
That is super interesting, thanks @Omnikar! In the interest of future searches, I'm gonna refer to this as "treating the stack as an array" or "applying uiua ops to the stack" 😄
